hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +50,123 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
     // sync the embedded group
     m_cl->setEmbeddedGroup(m_embed_group);
 
+    // check for collision warnings
+    checkCollisionWarnings(timestep);
+
     // set random grid shift
     m_cl->drawGridShift(timestep);
 
     // update cell list
     m_cl->compute(timestep);
 
+    // create auxillary array for rigid bodies
+    beginRigidBodyCollision(timestep);
+
+    // apply collisions
     rule(timestep);
+
+    // apply collisions to rigid bodies
+    finishRigidBodyCollision(timestep);
+    }
+
+void mpcd::CollisionMethod::checkCollisionWarnings(uint64_t timestep)
+    {
+    if (m_checked_collision_warnings)
+        {
+        return;
+        }
+    if (m_embed_group)
+        {
+        unsigned int N_tot = m_embed_group->getNumMembers();
+        ArrayHandle<unsigned int> h_embed_group(m_embed_group->getIndexArray(),
+                                                access_location::host,
+                                                access_mode::read);
+        ArrayHandle<Scalar4> h_vel_embed(m_pdata->getVelocities(),
+                                         access_location::host,
+                                         access_mode::read);
+        ArrayHandle<unsigned int> h_body_embed(m_pdata->getBodies(),
+                                               access_location::host,
+                                               access_mode::read);
+        ArrayHandle<unsigned int> h_rtag_embed(m_pdata->getRTags(),
+                                               access_location::host,
+                                               access_mode::read);
+
+        // check if some of the masses are less or equal to 0
+        bool invalid_mass = false;
+        bool central_interacting = false;
+        for (unsigned int idx = 0; idx < N_tot && !invalid_mass && !central_interacting; ++idx)
```

Comment:
I think you want to group like this so the loop will only terminate if both warnings have tripped. Otherwise, it will quit early when one trips.
```suggestion
        for (unsigned int idx = 0; idx < N_tot && !(invalid_mass && central_interacting); ++idx)
```